### PR TITLE
Misc: Remove pykdtree dependency, codecov config, SPEC 0 tooling and other changes

### LINKED ISSUE #1616: Two libraries used for KDTree in Parcels. Use only one?
Throughout the codebase, Parcels uses two different libraries for `kdtree`:

https://github.com/OceanParcels/parcels/blob/035159a1b4002bc1550cebd275f93c0adaa8513d/parcels/particleset.py#L18-L21

https://github.com/OceanParcels/parcels/blob/035159a1b4002bc1550cebd275f93c0adaa8513d/parcels/interaction/neighborsearch/kdtreeflat.py#L2

Perhaps we should standardise this and use only one of them? Since `scipy` is already a dependency, perhaps better to use `scipy.spatial.KDTree` throughout?

### PR DESCRIPTION
**Pull request by VeckoTheGecko**
Several small changes:
- doc: add release link to navbar
- doc: update warning message on documentation page (related to #1665)
- doc: clarify timedelta conversion in `tutorial_output` notebook
- Remove pykdtree dependency (fixes #1616)
- Update codecov config to enable PR comments
- Add reference file (along with check of that file) for compliance with SPEC 0 (completes #1658)

### CODE REVIEW

**Comment by VeckoTheGecko** [comment]
This is the output from the min-deps check workflow
```
Package           Required             Policy               Status
----------------- -------------------- -------------------- ------
python            3.10    (2021-10-04) 3.9     (2020-10-05) > (!)
cftime            1.6     (2022-03-03) 1.6     (2022-03-03) =
cgen              2020.1  (2020-02-07) 2020.1  (2020-02-07) =
dask              2022.8  (2022-08-05) 2022.8  (2022-08-05) =
matplotlib-base   3.5     (2021-11-18) 3.5     (2021-11-18) =
netcdf4           1.6     (2022-06-29) 1.6     (2022-06-29) =
numpy             1.23    (2022-06-23) 1.23    (2022-06-23) =
platformdirs      2.5     (2022-02-10) 2.5     (2022-02-10) =
psutil            5.9     (2021-12-30) 5.9     (2021-12-30) =
pymbolic          2022.1  (2022-05-09) 2022.1  (2022-05-09) =
pytest            7.1     (2022-03-14) 7.1     (2022-03-14) =
scipy             1.9     (2022-07-30) 1.9     (2022-07-30) =
trajan            0.1     (2022-11-22) 0.1     (2022-11-22) =
tqdm              4.64    (2022-04-04) 4.64    (2022-04-04) =
xarray            2022.6  (2022-07-23) 2022.6  (2022-07-23) =
zarr              2.12    (2022-06-23) 2.12    (2022-06-23) =
```

Currently the workflow gets run every 3 months. If a dependency in the reference file is out of date according to SPEC 0 it gets flagged in that table and CI gets failed.

File: ci/min-core-deps.yml
Comment (erikvansebille) [inline ci/min-core-deps.yml:1]: So we're making a new `ci`-directory in the root? Would this not fit better under `.github` or `tests`?

File: docs/documentation/index.rst
Comment (erikvansebille) [inline docs/documentation/index.rst:1]: Should we (soon) release v3.1.0 then? Otherwise readers may think it exists already

File: ci/min-core-deps.yml
Comment (VeckoTheGecko) [reply ci/min-core-deps.yml:1]: I can nest it in .github

Comment (erikvansebille) [reply ci/min-core-deps.yml:1]: Yes, I'd like to keep the root of the project repo as clean/empty as possible, so nesting under .github is a good idea

**Review by erikvansebille — APPROVED** [review]

**Pull request merged**